Remotion/Web/Core/UI/Controls/WebTreeNodeCollection.cs

```diff
@@ -23,6 +25,18 @@ namespace Remotion.Web.UI.Controls
   /// <summary> A collection of <see cref="WebTreeNode"/> objects. </summary>
   public class WebTreeNodeCollection : ControlItemCollection
   {
+    /// <summary>
+    /// Groups the top level <see cref="WebTreeNode"/>s by their <see cref="WebTreeNode.Category"/>
+    /// </summary>
+    /// <param name="webTreeNodes">The nodes of the <see cref="WebTreeView"/></param>
+    /// <returns></returns>
+    public static IReadOnlyList<WebTreeNode> GroupByCategory (WebTreeNodeCollection webTreeNodes)
+    {
+      var nodes = ArgumentUtility.CheckNotNullAndType<WebTreeNode[]> ("webTreeNodes", webTreeNodes.Cast<WebTreeNode>().ToArray());
+
+      return nodes.GroupBy (node => node.Category).SelectMany (node => node).ToArray();
```

Comment:
```suggestion
      ArgumentUtility.CheckNotNull ("webTreeNodes", webTreeNodes);

      return webTreeNodes.Cast<WebTreeNode>().GroupBy (node => node.Category).SelectMany (node => node).ToArray();
```

The argument type is already ensured via the WebTreeNodeCollection so we may do a hard cast.